packages/node/src/build.ts

```diff
@@ -143,14 +143,50 @@ async function compile(
   }
 
   let tsCompile: Register;
-  function compileTypeScript(path: string, source: string): string {
+  async function compileTypeScript(
+    path: string,
+    source: string
+  ): Promise<string> {
     const relPath = relative(baseDir, path);
     if (!tsCompile) {
+      // Find package.json to determine module type
+      const pathDir = join(workPath, dirname(path));
+      let pkg: { type?: string } = {};
+
+      // Check if we already have the package.json in cache
+      if (pkgCache.has(pathDir)) {
+        pkg = pkgCache.get(pathDir) || {};
+      } else {
+        // Load package.json and add to cache
+        try {
+          const pathToPkg = await walkParentDirs({
```

Comment:
Doesn't this logic already existing somewhere? I know we already support esm so seems like we could reuse that instead of walking the tree again.